### ISSUE TITLE
Improve method JSON descriptions

### ISSUE THREAD

**Issue by jasonpaulos** [algorand/pyteal#516]
Currently, the JSON description field for an ABI method is simply the method's Python docstring, with all newlines turned into spaces: https://github.com/algorand/pyteal/blob/a66907ef4341db8e595f97f4a53413d6338f6c74/pyteal/ast/subroutine.py#L630-L637

This can be improved in a few ways:
1. Keep filtering out single newlines (which are used to wrap a single line in the Python docstring), but treat double newlines as a newline in the JSON doc string. For example, given the following Python docstring:
```python
def example1():
    """This is an example docstring. This first line wraps since it's too
    long to fit in a single line of source code.
    
    This is a second line.
    """
    pass
```
We should produce this JSON string:
```json
"This is an example docstring. This first line wraps since it's too long to fit in a single line of source code.\nThis is a second line."
```
2. Right now, there is no way to populate the description fields of arguments or return values. I propose that we adopt some way to parse a Python docstring and extract information about these fields. For example, given the following Python docstring:
```python
def example2(a: abi.Uint8, b: abi.Uint32, *, output: abi.Uint64):
    """This is a second example docstring.
    
    Args:
        a: A description for the first argument.
        b: A description for the second argument.
    
    Returns:
        A description for the return value.
    """
    pass
```
We should produce the following method JSON:
```json
{
  "name": "example2",
  "args": [
    {
      "type": "uint8",
      "name": "a",
      "desc": "A description for the first argument."
    },
    {
      "type": "uint32",
      "name": "b",
      "desc": "A description for the second argument."
    }
  ],
  "returns": {
    "type": "uint64",
    "desc": "A description for the return value."
  },
  "desc": "This is a second example docstring."
}
```

**Comment by cusma** [comment]
Which `docstring` format should we use? I think the most popular in Python are:

- [Epytext](http://epydoc.sourceforge.net/manual-epytext.html)
- [reST](https://sphinx-rtd-tutorial.readthedocs.io/en/latest/docstrings.html#the-sphinx-docstring-format) used by Sphinx (and default in JetBrains PyCharm)
- [Goolge](https://github.com/google/styleguide/blob/gh-pages/pyguide.md#38-comments-and-docstrings) (also supported by Python IDEs like PyCharm)
- [NumPy](https://sphinxcontrib-napoleon.readthedocs.io/en/latest/example_numpy.html#example-numpy)

I personally find the Google docstring style guide the best one in terms of readability, and I think we are already using it in `pyteal`.

**Comment by jasonpaulos** [comment]
Good point, I forget there is more than just the Google format sometimes 😅 

I'd prefer supporting Google, but if we use a library like [docstring_parser](https://github.com/rr-/docstring_parser), it appears to support all 4 of those formats.
